Src/Idioms/EqualityAssertion.cs

```diff
@@ -0,0 +1,364 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture.Idioms
+{
+	/// <summary>
+	/// Represent an assertion which verifies that an equality method returns the correct value.
+	/// </summary>
+	public class EqualityAssertion : IdiomaticAssertion
+	{
+		private readonly Func<object, object, bool> comparer;
+		private IFixture fixture;
+
+		/// <summary>
+		/// Creates a new instance of EqualityAssertion which verified the Object.Equals method
+		/// </summary>
+		/// <param name="fixture">A composer which can create instances required to implement the idiomatic unit test.</param>
+		public EqualityAssertion(IFixture fixture)
+			: this(fixture, (x, y) => x.Equals(y))
```

Comment:
It would be more clear if you used an `IEqualityComparer` here. Additionally, we have used this in some [other](https://github.com/AutoFixture/AutoFixture/blob/d02f102301af95908e31dcb0e5c628abe48cbb8b/Src/Idioms/ConstructorInitializedMemberAssertion.cs#L47) [places](https://github.com/AutoFixture/AutoFixture/blob/d02f102301af95908e31dcb0e5c628abe48cbb8b/Src/Idioms/CopyAndUpdateAssertion.cs#L59) already, therefore it would be more consistent.